d2l-input-text-behavior.html

```diff
@@ -0,0 +1,108 @@
+<script>
```

Comment:
I moved the properties and functions shared by https://github.com/BrightspaceUI/text-input/blob/master/d2l-text-input.html and https://github.com/BrightspaceUI/textarea/blob/master/d2l-textarea.html into a behaviour for the two inputs to share once `textarea` is added here as well.